pkg-manager/resolve-dependencies/src/getCatalogSnapshots.ts

```diff
@@ -7,9 +8,18 @@ export function getCatalogSnapshots (resolvedDirectDeps: readonly ResolvedDirect
 
   for (const dep of catalogedDeps) {
     const snapshotForSingleCatalog = (catalogSnapshots[dep.catalogLookup.catalogName] ??= {})
+
     snapshotForSingleCatalog[dep.alias] = {
       specifier: dep.catalogLookup.specifier,
-      version: dep.version,
+
+      // Note: The version recorded here is used for lookups of this cataloged
+      // dependency in future installs. This should be computed the same as
+      // other version refs in the lockfile.
+      version: depPathToRef(dep.pkgId, {
+        alias: dep.alias,
+        realName: dep.name,
+        resolution: dep.resolution,
+      }),
```

Comment:
As noted in the PR description, this follows how we compute the version `ref` for `importers`.

https://github.com/pnpm/pnpm/blob/0406d4ad3d2a10326cd46ccc242aadad878a5ead/pkg-manager/resolve-dependencies/src/index.ts#L395-L399